nixos/modules/services/x11/extra-layouts.nix

```diff
@@ -0,0 +1,154 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  layouts = config.services.xserver.extraLayouts;
+
+  layoutOpts = {
+    options = {
+      description = mkOption {
+        type = types.str;
+        description = "A short description of the layout.";
+      };
+
+      languages = mkOption {
+        type = types.listOf types.str;
+        description =
+        ''
+          A list of languages provided by the layout.
+          (Use ISO 639-2 codes, for example: "eng" for english)
+        '';
+      };
+
+      filepath = mkOption {
```

Comment:
One last nitpick (sorry, last one for sure this time, from me at least): I'd prefer `layoutFile` over `filepath`, the former is more in the style of other NixOS options.

Reply:
Ok, sure.